### PR TITLE
[IMPROV] Quality of Life improvments for the devcontainer

### PR DESCRIPTION
**Pull request by AlexVCaron**
## Type of improvement

**If submitting a new module or fixing a bug, please use the appropriate template.**

- [x] Documentation
- [x] Development tools (e.g. linter, formatter, etc.)
- [x] Development container
- [ ] Global update (please specify)
- [ ] Other (please specify)

## Describe your improvement

Quality of life improvements for the development container :

- Auto-populate `NFCORE_*_GIT_REMOTE` and `NFCORE_*_BRANCH`. No need to add `--git-remote` or `--branch` to the `nf-core` commands inside the container anymore
- Reinstate the `.venv` python environment inside the workspace. We'll ignore it everywhere we can, but it must live there for `vscode` to detect it correctly.
- Add `user profile` persistence (command history should carry between rebuilds). Requires to rebuild the container once, so the volumes get instantiated and linked to the container by docker
- Place the `.nf-test` folder living inside the workspace in its own volume, so we can delete it easily if the container runs out of space and does want to start
- Add `NiiVue` to the extensions stack. Enjoy looking at images from inside `vscode` now ! Now, when you click on a nifti in the explorer, it opens in a 3D (even 4D !) viewer in the editor. The 3D mean intensity projection view is quite nice, ! But it does handle all space transformations perfectly yet, doesn't have peaks, nor fodf, nor tensors and is limited in sense of colormaps. Still, very nice to have in the container.
- Split devcontainer external configuration (not much, but it's clearer what the content of the `.devcontainer` folder does)

## Describe how to test your improvement

Remove all containers and volumes related to the `nf-scil` devcontainer from docker. All resources should be prefixed with `nf-scil`. You won't loose your work, but you'll loose any cache living outside the workspace folder.

## Checklist before requesting a review

- [x] Ensure the syntax is correct (**EditorConfig** and **Prettier** must pass)
- [x] Run the test suites if your changes affect any module
- [x] Regenerate the **Poetry** lock file if you have updated the dependencies
- [x] Ensure the documentation is up-to-date

### CODE REVIEW

**Comment by AlexVCaron** [comment]
Still need to update all documentation :stuck_out_tongue_winking_eye:

**Comment by AlexVCaron** [comment]
Documentation should be updated !

File: nf-test.config [hunk 1]
```diff
@@ -10,4 +10,6 @@ config {
 
     // run all test with the defined docker profile from the main nextflow.config
     profile ""
-}
\ No newline at end of file
+
+    ignore [".venv/*"]
```
Comment (gagnonanthony) [inline nf-test.config:14]: It seems the ignore flag is not available in the current release of `nf-test` (0.8.4). When I try to run a test for a module, it outputs this error message: 
```
Error: Syntax errors in nf-test config file: groovy.lang.MissingPropertyException: No such property: ignore for class: com.askimed.nf.test.config.Config
```
I quickly went through the nf-test repo, and it seems the ignore was commited last month for the 0.9.0 release. I also might be wrong 😆

**Review by gagnonanthony — APPROVED** [review]
Wow, that's really nice! 💯 I just tested it, and it truly makes it much easier to develop within the container (even on Mac...). I encountered only one bug, which I describe below. Otherwise, looks pretty good to me!

**Review by arnaudbore — APPROVED** [review]
GTG

**Pull request merged**